examples/all-clusters-app/nxp/mw320/main.cpp

```diff
@@ -172,7 +172,7 @@ void InitOTARequestor(void)
     // TODO: instatiate and initialize these values when QueryImageResponse tells us an image is available
     // TODO: add API for OTARequestor to pass QueryImageResponse info to the application to use for OTADownloader init
     // OTAImageProcessor ipParams;
-    // ipParams.imageFile = CharSpan("dnld_img.txt");
```

Comment:
We need to find a way to make this do the right thing, or otherwise break.

Reply:
I wonder if we can have enable-if magic on a deleted constructor when type of span is char, or const char

Reply:
I am not sure I can do something actionable here. At least testing will find this kind of errors somewhat fast (especially for hard-coded strings). However not having this compile is likely harder and a separate task).

Reply:
https://github.com/project-chip/connectedhomeip/pull/30400